src/DataGenerator.h

```diff
@@ -0,0 +1,188 @@
+#ifndef DATAGENERATOR_H
+#define DATAGENERATOR_H
+
+// will be using pugiXML for parsing and generating
+// git clone https://github.com/zeux/pugixml.git
+// forked repo and uncomment in pugixml/src/pugiconfig.hpp
+// line 48 #define PUGIXML_HEADER_ONLY
+
+#include "../extsrc/pugixml/src/pugixml.hpp"
+#include "AtomicMCDAObject.h"
+#include "Criterion.h"
+#include "Performance.h"
+#include "PerformanceTable.h"
+#include <iostream>
+#include <tuple>
+
+class DataGenerator : public AtomicMCDAObject {
+public:
+  DataGenerator(){};
+
+  /**
+   * Generate a random dataset and put it in a xml file
+   *
+   * @param nb_criteria number of criteria
+   * @param nb_alternative number of alternatives
+   * @param nb_categories number of categories
+   * @param datasetName name of that dataset
+   * @param overwrite bool that refers to if we want to overwrite file
+   * @param seed seed to control random generation
+   *
+   * @return a saved file of the name datasetName.xml in data/ or default
+   * generated name
+   */
+  void datasetGenerator(int nb_criteria, int nb_alternative, int nb_categories,
+                        std::string datasetName = "", bool overwrite = 1,
+                        bool changeSeed = 0);
+
+  /**
+   * Generate a random model and puts it in a xml file
+   *
+   * @param nb_criteria number of criteria
+   * @param nb_categories number of categories
+   * @param modelName name of that model
+   * @param overwrite bool that refers to if we want to overwrite file
+   * @param seed seed to control random generation
+   *
+   * @return a saved file of the name nametName.xml in data/ or default
+   * generated name
+   */
+  void modelGenerator(int nb_criteria, int nb_categories, std::string modelName,
+                      bool overwrite = 1, bool changeSeed = 0);
+
+  /**
+   * Get data from xml file
+   *
+   * @param fileName filename
+   *
+   * @return a type for model usage
+   *
+   */
+  // void loadDataset(std::string fileName);
```

Comment:
Are those 2 functions something you will implement later or where they left here by mistake?